doubly_linked_list/doubly_linked_list.py

```diff
@@ -3,90 +3,155 @@
+	have a next node it is point to."""
+	def insert_after(self, value):
+		current_next = self.next
+		self.next = ListNode(value, self, current_next)
+		if current_next:
+			current_next.prev = self.next
+
+	"""Wrap the given value in a ListNode and insert it
+	before this node. Note that this node could already
+	have a previous node it is point to."""
+	def insert_before(self, value):
+		current_prev = self.prev
+		self.prev = ListNode(value, current_prev, self)
+		if current_prev:
+			current_prev.next = self.prev
+
+	"""Rearranges this ListNode's previous and next pointers
+	accordingly, effectively deleting this ListNode."""
+	def delete(self):
+		if self.prev:
+			self.prev.next = self.next
+		if self.next:
+			self.next.prev = self.prev
 
 
 """Our doubly-linked list class. It holds references to
 the list's head and tail nodes."""
 
 
 class DoublyLinkedList:
-    def __init__(self, node=None):
-        self.head = node
-        self.tail = node
-        self.length = 1 if node is not None else 0
-
-    def __len__(self):
-        return self.length
-
-    """Wraps the given value in a ListNode and inserts it 
-    as the new head of the list. Don't forget to handle 
-    the old head node's previous pointer accordingly."""
-    def add_to_head(self, value):
-        pass
-
-    """Removes the List's current head node, making the
-    current head's next node the new head of the List.
-    Returns the value of the removed Node."""
-    def remove_from_head(self):
-        pass
-
-   """Wraps the given value in a ListNode and inserts it 
-    as the new tail of the list. Don't forget to handle 
-    the old tail node's next pointer accordingly."""
-    def add_to_tail(self, value):
-        pass
-
-    """Removes the List's current tail node, making the 
-    current tail's previous node the new tail of the List.
-    Returns the value of the removed Node."""
-    def remove_from_tail(self):
-        pass
-
-    """Removes the input node from its current spot in the 
-    List and inserts it as the new head node of the List."""
-    def move_to_front(self, node):
-        pass
-
-    """Removes the input node from its current spot in the 
-    List and inserts it as the new tail node of the List."""
-    def move_to_end(self, node):
-        pass
-
-    """Removes a node from the list and handles cases where
-    the node was the head or the tail"""
-    def delete(self, node):
-        pass
-        
-    """Returns the highest value currently in the list"""
-    def get_max(self):
-        pass
+	def __init__(self, node=None):
+		self.head = node
+		self.tail = node
+		self.length = 1 if node is not None else 0
+
+	def __str__(self):
+		current_node = self.head
+		values = ""
+		while current_node:
+			if current_node is self.tail:
+				values += str(current_node.value)
+			else:
+				values += str(current_node.value) + ", " 
+			current_node = current_node.next
+		return f'DoublyLinkedList: values: [{values}], length: {self.length}'
```

Comment:
nice! We don't even do the string method in the solution. Bonus points!
![10pts](https://user-images.githubusercontent.com/5281855/66864391-e9763500-ef49-11e9-8185-bd891ebc1892.gif)